Kinesis docs update (#2685)
SumoLogic/sumologic-documentation: docs/send-data/collect-from-other-data-sources/amazon-cloudwatch-logs/collect-with-lambda-function.md

sidebar_label: Collect Logs using a Lambda Function
description: Learn how to collect Amazon CloudWatch Logs with a Lambda function.
---

> import useBaseUrl from '@docusaurus/useBaseUrl';

Reviewer: ```suggestion

import useBaseUrl from '@docusaurus/useBaseUrl';
```

:::note
We strongly recommend the alternative collection process described on [AWS Kinesis Firehose for Logs Source](/docs/send-data/hosted-collectors/amazon-aws/aws-kinesis-firehose-logs-source/), which is more robust and reliable, where you do not need to manage resources. Lambda based collection methods are limited by time out, concurrency, and memory limits.
:::

> :::caution
> This method is highly not recommended as it does not handle failures and may lead to data loss. In addition, the data format generated by this method might be incompatible with some applications in the **App catalog** and thus requires customization of those apps.

Reviewer: ```suggestion
:::caution
This method is highly not recommended as it does not handle failures and may lead to data loss. Additionally, the data format generated by this method might be incompatible with some applications in our **App Catalog** and thus requires customization of those apps.
```

:::

> The following instructions tell you to how download and configure an AWS Lambda function for Amazon CloudWatch Logs and send them to Sumo Logic.

Reviewer: ```suggestion
The following instructions describe how to download and configure an AWS Lambda function for Amazon CloudWatch Logs and send them to Sumo Logic.
```

1. On the **Create Functions** page, select **Use a blueprint** option.
1. In the **Basic information** section, enter the following.<br/><img src={useBaseUrl('img/send-data/basic-information.png')} style={{border: '1px solid black'}} alt="basic-information" width="700"/>
    * **Blueprint name**. Enter `sumologic` and select **Send Cloudwatch logs to Sumologic**.
    * **Function name**. Enter a name for the function. Default is `sumologic-cwl-lambda`.
    * **Role**. Choose one of the following options.
      * **Use an existing role.** If you have any appropriate roles, you can select one.
      * **Create a new role with basic Lambda permissions**. If you select this option, you can continue without choosing any policy templates—it will create a role with basic Lambda execution privileges by default.
      * **Create a new role from AWS policy templates**.
        * **Role Name**. Enter a name for the role.
        * **Policy templates**. Choose the template and make sure that the policy has following permissions `logs:CreateLogGroup, logs:CreateLogStream, logs:PutLogEvents` so that Lambda can push the logs to Cloudwatch. 

> 

Reviewer: ```suggestion
```

1. In the **CloudWatch Logs trigger** section, you can create a trigger, or click **Remove** if you prefer to create it later in [Step 3](#step-3-assign-cloudwatch-log-groups-to-your-lambda-function). To create the trigger:<br/><img src={useBaseUrl('img/send-data/cloudWatch-logs-trigger.png')} style={{border: '1px solid black'}} alt="cloudWatch-logs-trigger" width="700"/>
    * **Log Group**. Select the log group that serves as the event source. Events sent to the log source will trigger your Lambda function. 
    * **Filter Name**. Enter a filter name.

>     * **Filter Pattern (Optional)**. May be left blank. For information about AWS filter patterns, see [Filter and Pattern Syntax](http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html) in AWS help.

Reviewer: ```suggestion
    * **Filter Pattern (Optional)**. May be left blank. For information about AWS filter patterns, see [Filter and Pattern Syntax](http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html) in AWS Help.
```

    * **ENCODING (Optional)**. Use when decoding CloudWatch log events, default is **utf-1**.
    * **SOURCE_CATEGORY_OVERRIDE (Optional)**. Override `_sourceCategory` value configured for the HTTP source.  
    * **SOURCE_HOST_OVERRIDE (Optional)**. Override `_sourceHost` value configured for the HTTP source.
    * **SOURCE_NAME_OVERRIDE (Optional)**. Override `_sourceName` value configured for the HTTP source.
1. Click **Create Function.** A success message will be displayed.<br/><img src={useBaseUrl('img/send-data/success.png')} style={{border: '1px solid black'}} alt="success" width="700"/>

## Step 3: Assign CloudWatch Log Groups to Your Lambda Function

:::info
You will need at least one CloudWatch Log Group to assign to your Lambda function. For details on how to create a CloudWatch Log Group, see [Create a CloudWatch Log Group](http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/Create-Log-Group.html).
:::

:::note

> If you have more than one CloudWatch Log Group to assign to your Lambda function, repeat the below steps for each Log Group, else skip this step.

Reviewer: ```suggestion
If you have more than one CloudWatch Log Group to assign to your Lambda function, repeat the below steps for each Log Group. Otherwise, skip this step.
```
